apps/main/src/loan/components/MarketInformationComp.tsx

```diff
@@ -1,52 +1,107 @@
+import { BandsChart } from '@/llamalend/features/bands-chart/BandsChart'
+import { useBandsData } from '@/llamalend/features/bands-chart/hooks/useBandsData'
 import { BandsComp } from '@/loan/components/BandsComp'
 import ChartOhlcWrapper from '@/loan/components/ChartOhlcWrapper'
 import DetailInfoAddressLookup from '@/loan/components/LoanInfoLlamma/components/DetailInfoAddressLookup'
 import LoanInfoParameters from '@/loan/components/LoanInfoLlamma/LoanInfoParameters'
 import { SubTitle } from '@/loan/components/LoanInfoLlamma/styles'
+import networks from '@/loan/networks'
 import type { ChainId, Llamma } from '@/loan/types/loan.types'
 import { Stack, useTheme } from '@mui/material'
+import { useConnection } from '@ui-kit/features/connect-wallet'
 import { useUserProfileStore } from '@ui-kit/features/user-profile'
+import { useNewBandsChart } from '@ui-kit/hooks/useFeatureFlags'
 import { t } from '@ui-kit/lib/i18n'
 import { SizesAndSpaces } from '@ui-kit/themes/design/1_sizes_spaces'
 
 const { Spacing } = SizesAndSpaces
 
 type MarketInformationCompProps = {
   llamma: Llamma | null
-  llammaId: string
+  marketId: string
   chainId: ChainId
   chartExpanded: boolean
   page?: 'create' | 'manage'
 }
 
+const EMPTY_BANDS_BALANCES: never[] = []
+
 /**
  * Reusable component for OHLC charts, Bands, and market parameters. For /create and /manage pages.
  */
 export const MarketInformationComp = ({
   llamma,
-  llammaId,
+  marketId,
   chainId,
   chartExpanded,
   page = 'manage',
 }: MarketInformationCompProps) => {
+  const { llamaApi: api } = useConnection()
   const theme = useTheme()
+  const newBandsChartEnabled = useNewBandsChart()
   const isAdvancedMode = useUserProfileStore((state) => state.isAdvancedMode)
+  const collateralTokenAddress = llamma?.coinAddresses[1]
+  const borrowedTokenAddress = llamma?.coinAddresses[0]
+  const {
+    chartData,
+    userBandsBalances,
+    oraclePrice,
+    isLoading: isBandsLoading,
+    isError: isBandsError,
+  } = useBandsData({
+    chainId,
+    marketId,
+    api,
+    collateralTokenAddress,
+    borrowedTokenAddress,
+  })
+  const collateralToken =
+    llamma && collateralTokenAddress
+      ? {
+          symbol: llamma.collateralSymbol,
+          address: collateralTokenAddress,
+          chain: networks[chainId].id,
```

Comment:
why do we need the `networks` mapping for this?
```suggestion
          chain: chainId,
```